psi4/src/psi4/libfunctional/LibXCfunctional.cc

```diff
@@ -194,12 +195,22 @@ std::shared_ptr<Functional> LibXCFunctional::build_worker() {
     func->set_meta(meta_);
     func->set_lsda_cutoff(lsda_cutoff_);
     func->set_meta_cutoff(meta_cutoff_);
+    func->set_density_cutoff(density_cutoff_);
     func->exc_ = exc_;
     func->vxc_ = vxc_;
     func->fxc_ = fxc_;
 
     return static_cast<std::shared_ptr<Functional>>(func);
 }
+void LibXCFunctional::set_density_cutoff(double cut) {
+    density_cutoff_ = cut;
+    if (density_cutoff_ > 0){
+        xc_func_set_dens_threshold(xc_functional_.get(),cut);
+    }
+}
+double LibXCFunctional::query_density_cutoff() {
```

Comment:
As far as I can see the only reason its here is if you don't trust the interface so this is a testing function which is better handled by an actual test (setting the density tolerance).

Reply:
The main use is in the `SuperFunctional::print_density_threshold` function https://github.com/psi4/psi4/pull/1860/files#diff-7c91ae12434f3595779f6de77e652235R407  to also print the internal defaults used by LibXC and not just the user specified values. (with the caveat that `libxc_xc_func` functionals do not report threshold for their functional components , see CAM-B3LYP issue in the main post)